packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -266,6 +273,7 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
     /// Rebuilds the table whenever the tree structure has been updated
     setState(() {
       selectedNode = node;
+      selectedNodeIndex = items.indexOf(selectedNode);
```

Comment:
This can become expensive to compute for large list sizes.

Some of the tables are already building the nodes by a particular index.

I suggest either making the index a property of the node or passing that index in to the _onItemPressed method.

One way you can do this without creating any new fields on the node object is by currying: change this method to return a `void Function(T node)` instead of the current `void` return type:

```
void Function(T) _onItemPressed(int index) {
  return (T node) {
    ...The current _onItemPressed function
  };
}
```

You then change how this is used at the call site below:

```
Widget _buildRow(context, linkedScrollControllerGroup, int index) {
  ...
  return TableRow(
    ...
    onPressed: _onItemPressed(index),
    ...
  );
}
```

Reply:
Added the index by passing it in from buildRow